docs/src/devdocs/FEValues.md

```diff
@@ -26,3 +26,17 @@ Ferrite.FunctionValues
   the jacobian, $\partial M/\partial x$, is calculated. 
 * In the second part of `reinit!`, The `MappingValues`, containing sufficient information for the current quadrature point, is given to `apply_mapping!`. 
   This allows the shape values and gradients stored in `FunctionValues`, to be mapped to the current cell by calling `apply_mapping!`.
+
+## Custom FEValues
+Custom FEValues, `fe_v`, should normally implement the `reinit!` method.
+Additionally, for normal functionality the `getnquadpoints` should be implemented.
+Note that asking for the `n`th quadrature point must be inside array bounds if 
+`1<=n<:getnquadpoints(fe_v)`
+(`checkquadpoint` can, alternatively, be dispatched to check that `n` is inbounds.)
+
+Supporting `function_value`, `function_gradient`, `function_symmetric_gradient`, `function_divergence`, and `function_curl`,
```

Comment:
I think this is not correct. We should rephrase that we have more requirements here (like for example the fields need to have specific names in master). Or we add more functions to query these, which I think is better. Maybe we can think about a trait system for cell values, too?

Reply:
Maybe I misunderstand, but there should be no requirements on specific field names left, see the example for `SimpleCellValues` that fulfill this specification. Or if you meant something else, could you elaborate?

Reply:
Maybe I missed something, but if I rename `dNdx` to `fun_by_x`in `SimpleCellValues`, then function* won't work anymore right?

Reply:
Yes it will, the `function_value` just calls `shape_value`. (In this branch, the built-in cell values and face values don't have this field)

On master, there were a few (IMO) antipatterns in `common_values.jl` dispatching on `AbstractValue` inputs and expecting certain fields to exist, but these are now removed.